open-widget-framework/test/defaults_test.js

```diff
@@ -0,0 +1,389 @@
+    expect(wrap.state('addWidgetForm')).to.equal(false)
+    expect(wrap.state('editWidgetForm')).to.equal(null)
+  })
+
+  it('toggleEditMode toggles the editMode in state', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    instance.toggleEditMode()
+
+    expect(wrap.state('editMode')).to.equal(true)
+    expect(wrap.state('addWidgetForm')).to.equal(false)
+    expect(wrap.state('editWidgetForm')).to.equal(null)
+
+    instance.toggleEditMode()
+
+    expect(wrap.state('editMode')).to.equal(false)
+    expect(wrap.state('addWidgetForm')).to.equal(false)
+    expect(wrap.state('editWidgetForm')).to.equal(null)
+  })
+
+  it('editWidget sets editWidgetForm to an id', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    const dummyId = 3
+    instance.editWidget(dummyId)
+
+    expect(wrap.state('editWidgetForm')).to.equal(dummyId)
+    expect(wrap.state('addWidgetForm')).to.equal(false)
+    expect(wrap.state('editMode')).to.equal(true)
+
+    instance.editWidget(dummyId + 1)
+
+    expect(wrap.state('editWidgetForm')).to.equal(dummyId + 1)
+    expect(wrap.state('addWidgetForm')).to.equal(false)
+    expect(wrap.state('editMode')).to.equal(true)
+
+  })
+
+  // Method + render: Ensure that method calls cause the appropriate components to be rendered
+  it('calls renderList and no other form render functions by default', () => {
+    resetSpyHistory()
+    mount(<ListWrapper {...dummyProps}/>)
+
+    expect(dummyProps.renderNewWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderEditWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderList.callCount).to.equal(1)
+  })
+
+  it('addWidget causes just renderNewWidgetForm to be called', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    resetSpyHistory()
+    instance.addWidget()
+
+    expect(dummyProps.renderNewWidgetForm.callCount).to.equal(1)
+    expect(dummyProps.renderEditWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderList.callCount).to.equal(1)
+  })
+
+  it('editWidget causes just renderEditWidgetForm to be called', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    resetSpyHistory()
+    instance.editWidget(dummyId)
+
+    expect(dummyProps.renderNewWidgetForm.callCount).to.equal(0)
+    expect(dummyProps.renderEditWidgetForm.withArgs(dummyId).callCount).to.equal(1)
+    expect(dummyProps.renderList.callCount).to.equal(1)
+  })
+
+  it('toggleEditWidget causes renderAddWidgetButton to be called', () => {
+    const wrap = mount(<ListWrapper {...dummyProps}/>)
+    const instance = wrap.instance()
+    resetSpyHistory()
+    const renderAddWidgetButtonSpy = sinon.spy(instance, 'renderAddWidgetButton')
+    instance.toggleEditMode()
+
+    expect(renderAddWidgetButtonSpy.callCount).to.equal(1)
```

Comment:
Can we also assert that the output of `renderAddWidgetButton` appears in the `wrap` rendered output?